py_ecc/optimized_bn128/optimized_curve.py

```diff
@@ -79,6 +79,9 @@ def is_on_curve(pt: Optimized_Point3D[Optimized_Field], b: Optimized_Field) -> b
 def double(
     pt: Optimized_Point3D[Optimized_Field],
 ) -> Optimized_Point3D[Optimized_Field]:
+    if is_inf(pt):
+        return pt
+
     x, y, z = pt
```

Comment:
I think we can also return early when `y == y.__class__.zero()` but I would need someone to verify that as no existing test hits this line when I made that change.

source: https://www.nayuki.io/page/elliptic-curve-point-addition-in-projective-coordinates

> Point doubling
...
If y=0, then 2P=O.